core/src/components/datetime/datetime.tsx

```diff
@@ -2071,20 +2079,39 @@ export class Datetime implements ComponentInterface {
       </ion-popover>,
     ];
   }
-  private renderCalendarViewHeader(mode: Mode) {
+  private renderCalendarViewHeader() {
     const hasSlottedTitle = this.el.querySelector('[slot="title"]') !== null;
     if (!hasSlottedTitle && !this.showDefaultTitle) {
       return;
     }
 
+    const { activeParts, selectedDatesLabelFormatter } = this;
+    const isArray = Array.isArray(activeParts);
+
+    let headerText;
+    if (isArray && activeParts.length !== 1) {
+      const defaultText = `${activeParts.length} days`;
+      if (selectedDatesLabelFormatter !== undefined) {
+        try {
+          headerText = selectedDatesLabelFormatter(activeParts.length);
+        } catch (e) {
+          printIonError('Exception in provided `selectedDatesLabelFormatter`: ', e);
+          headerText = defaultText; // fallback
+        }
+      } else {
+        headerText = defaultText;
+      }
+    } else {
+      // for exactly 1 day selected (multiple set or not), show a formatted version of that
+      headerText = getMonthAndDay(this.locale, isArray ? activeParts[0] : activeParts);
+    }
+
     return (
       <div class="datetime-header">
         <div class="datetime-title">
           <slot name="title">Select Date</slot>
         </div>
-        {mode === 'md' && !this.multiple && (
-          <div class="datetime-selected-date">{getMonthAndDay(this.locale, this.activeParts as DatetimeParts)}</div>
-        )}
+        <div class="datetime-selected-date">{headerText}</div>
```

Comment:
I didn't think of this aspect during the design phase, but should we allow `selectedDatesLabelFormatter` to return undefined to disable the rendering of this space?

Then this area could be conditionally rendered:
```jsx
{headerText && (<div class="datetime-selected-date">{headerText}</div>}
```

Reply:
I don't see why not 👀